### ISSUE TITLE
Long-running app architecture

### ISSUE THREAD

**Issue by harrywincup** [skpm/sketch-module-web-view#143]
Hi there,

### TLDR Question
How do I avoid all of my plugin code (and all imported modules) being re-run every time an action handler is called by Sketch? 

### Context
I am creating a `BrowserWindow` inside my `Startup` action handler, whose index.html initializes an Elm app for the UI inside the webview. The only menu command I have is Show/hide, whose handler uses `Remote.getWebview` to find the existing window and `show()`/`hide()` it accordingly or create a new one. Works as expected.

<img width="188" alt="image" src="https://user-images.githubusercontent.com/44423/76550967-daf56080-648a-11ea-9810-0e74ae20a0c8.png">

I notice however that each time I run my "Toggle UI" command from Sketch, the entire Javascript file that contains the handler and all imports are re-run, which means there can't be _any_ state in any of those files because it'll all be re-created each time.

I keep seeing people talk about using a Fiber, but i'm not sure if that's what I need (does BrowserWindow already have a Fiber going?) or how to use it if I do because I don't have any "longRunningTask" except wanting to keep all my code loaded for the duration of Sketch usage like I would with a regular web app. Is it possible to do that and if so what is the recommended architecture?

Hopefully somebody can point me in the right direction please. Thanks!

(Apologies if this is not quite the right place, or if it's been answered before. I did read all the historical issues but didn't see an answer. Closest I could find was https://github.com/skpm/sketch-module-web-view/issues/9 but all the links 404 :()

**Comment by harrywincup** [comment]
Update to this – I've removed state management on the "backend" plugin side and moved everything into the UI. Now, the Sketch Action handlers immediately send any data they fetch to the BrowserWindow where the Elm app does all its own state management. 

Of course, this doesn't help if you were trying something similar to my initial setup, but I had to move on and this works well for me, so apologies for not having a better answer.
